src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationNameCompletionProvider.cs

```diff
@@ -62,21 +62,31 @@ internal override bool IsInsertionTrigger(SourceText text, int insertedCharacter
             completionContext.SuggestionModeItem = CommonCompletionItem.Create(CSharpFeaturesResources.Name, CompletionItemRules.Default);
         }
 
-        private ImmutableArray<ImmutableArray<string>> GetBaseNames(SemanticModel semanticModel,  NameDeclarationInfo nameInfo)
+        private ImmutableArray<ImmutableArray<string>> GetBaseNames(SemanticModel semanticModel, NameDeclarationInfo info)
         {
-            if (nameInfo.Alias != null)
+            ImmutableArray<ImmutableArray<string>> baseNames = default;
+            if (info.Aliases != null)
             {
-                return NameGenerator.GetBaseNames(nameInfo.Alias);
+                baseNames = info.Aliases.ToSet().SelectMany(alias => NameGenerator.GetBaseNames(alias)).ToImmutableArray();
+            }
+            else if (info.Types != null)
+            {
+                baseNames = info.Types
+                    .Where(IsValidType)
+                    .ToSet()
+                    .Select(t => UnwrapType(t, semanticModel.Compilation))
+                    .SelectMany(t => NameGenerator.GetBaseNames(t))
+                    .ToImmutableArray();
             }
 
-            if (!IsValidType(nameInfo.Type))
+            if (info.ParameterNames != null)
```

Comment:
It's an ImmutableArray, so it'll never be null :-(

Where's that warning when you need it.